Modules/Scripted/SampleData/SampleData.py

```diff
@@ -684,6 +692,14 @@ def downloadFromSource(self, source, maximumAttemptsCount=3):
                     continue
                 resultFilePaths.append(filePath)
 
+                # Special case for backward compatibility for loading scene and zip files when loadFileType is not specified
```

Comment:
I suggest to keep the logic  in the `SampleDataSource` and update the code as described in https://github.com/Slicer/Slicer/pull/7479/files#r1428946872

Reply:
Does the comment mean that if the filetype is explicitly set to `SceneFile` or `ZipFile`, the behavior will be different ? 

I would suggest to have the same behavior .. may be the comment should instead be:

```

                # Special behavior to handle "ZipFile" and "SceneFile"
```